src/scaling/scaling.jl

```diff
@@ -92,9 +92,9 @@ coordlookup(r::UnitRange, x) = x - r.start + oneunit(eltype(r))
 # coordlookup(i::Bool, r::AbstractRange, x) = i ? coordlookup(r, x) : convert(typeof(coordlookup(r,x)), x)
 coordlookup(r::StepRange, x) = (x - r.start) / r.step + oneunit(eltype(r))
 
-coordlookup(r::StepRangeLen, x) = (x - first(r)) / step(r) + oneunit(eltype(r))
-boundstep(r::StepRangeLen) = 0.5*step(r)
-rescale_gradient(r::StepRangeLen, g) = g / step(r)
+coordlookup(r::AbstractRange, x) = (x - first(r)) / step(r) + oneunit(eltype(r))
+boundstep(r::AbstractRange) = 0.5*step(r)
+rescale_gradient(r::AbstractRange, g) = g / step(r)
```

Comment:
Are there potentially more instances that need a consistent change? Something like `rescale_hessian` for instance?

Reply:
Hmm. AFAICT no, `rescale_hessian_components`, for example, uses `steps`, which accepts any `AbstractRange` subtype. We should not have changed any behaviour. It presumably can't hurt to test the hessian scaling also though.

Reply:
Generally, I think you have found an oversight. #165 was marked as resolved by #226. So it indeed could be that this had been fixed in some places but not all.

Reply:
There are now tests for hessians specifically.. I'm not sure if they are useful tests.

tests/scaling.jl also tests for iteration behaviour and other float types when interpolating; but I don't think we especially want to clone those also for LinRange